packages/nx/src/project-graph/plugins/isolation/plugin-pool.ts

```diff
@@ -22,7 +22,16 @@ const cleanupFunctions = new Set<() => void>();
 
 const pluginNames = new Map<ChildProcess, string>();
 
-const MAX_MESSAGE_WAIT = 1000 * 60 * 5; // 5 minutes
+const MAX_MESSAGE_WAIT = (() => {
+  const envValue = process.env.NX_PLUGIN_MESSAGE_TIMEOUT;
+  if (envValue) {
+    if (envValue === '0' || envValue === 'false') {
+      return undefined;
+    }
+    return parseDuration(envValue);
+  }
```

Comment:
Let's keep it simple by having `NX_PLUGIN_NO_TIMEOUTS=true` to disable this timeout. No further customization for now